cmd/common.go

```diff
@@ -138,17 +139,32 @@ func syncMain(filenames []string, dry bool, parallelism, delay int) error {
 }
 
 func kongVersion(config utils.KongClientConfig) (semver.Version, error) {
+
+	var version string
 	client, err := utils.GetKongClient(config)
 	if err != nil {
 		return semver.Version{}, err
 	}
-
 	root, err := client.Root(nil)
 	if err != nil {
-		return semver.Version{}, err
-	}
-
-	v, err := utils.CleanKongVersion(root["version"].(string))
+		// try with workspace path
```

Comment:
Please guard this block with another if:
```go
if config.Workspace != ""
```

If workspace is not provided, then return back an error.